internal/controllers/user_devices_controller_test.go

```diff
@@ -1012,6 +1015,53 @@ func (s *UserDevicesControllerTestSuite) TestPostRefreshSmartCarRateLimited() {
 	}
 }
 
+func (s *UserDevicesControllerTestSuite) TestGetUserDevicesByIntegration() {
+	integration := test.BuildIntegrationGRPC(constants.TeslaVendor, 0, 0)
+	s.deviceDefSvc.EXPECT().GetIntegrationByID(gomock.Any(), constants.TeslaVendor).Return(integration, nil)
+
+	mockAuthCode := "Mock_fd941f8da609db8cd66b1734f84ab289e2975b1889a5bedf478f02cf0cc4"
+	mockRedirectURI := "https://mock-redirect.test.dimo.zone"
+	mockRegion := "na"
+
+	mockAuthCodeResp := &services.TeslaAuthCodeResponse{
+		AccessToken:  "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c",
+		RefreshToken: "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.UWfqdcCvyzObpI2gaIGcx2r7CcDjlQ0IzGyk8N0_vqw",
+		IDToken:      "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.ouLgsgz-xUWN7lLuo8qE2nueNgJIrBz49QLr_GLHRno",
+		ExpiresIn:    int(time.Now().Add(time.Hour * 1).Unix()),
+		State:        "pmXw63l3ax",
+		TokenType:    "",
+	}
+	s.teslaSvc.EXPECT().CompleteTeslaAuthCodeExchange(mockAuthCode, mockRedirectURI, mockRegion).Return(mockAuthCodeResp, nil)
+
+	resp := []services.TeslaVehicle{
+		{
+			ID:        11114464922222,
+			VehicleID: 44444699777777,
+			VIN:       "1GBGC24U93Z337558",
+		},
+		{
+			ID:        22222464999999,
+			VehicleID: 44444699000000,
+			VIN:       "WAUAF78E95A553420",
+		},
+	}
+	s.teslaSvc.EXPECT().GetVehicles(mockAuthCodeResp.AccessToken, mockRegion).Return(resp, nil)
+
+	request := test.BuildRequest("POST", "/user/devices/"+constants.TeslaVendor, fmt.Sprintf(`{
+		"authorizationCode": "%s",
+		"redirectUri": "%s",
+		"region": "na"
+	}`, mockAuthCode, mockRedirectURI))
+	response, _ := s.app.Test(request)
+
+	assert.Equal(s.T(), fiber.StatusOK, response.StatusCode)
```

Comment:
fun fact: suites have an `Assert` and `Require` function. Both version behave completely the same so there is no reason to change what you have here just sharing.
```suggestion
	s.Assert().Equal(fiber.StatusOK, response.StatusCode)
```